IllegalArgumentException when starting a payara micro on Windows via cli with --rootDir or --domainConfig

starting a micro instance under Windows like this
    `java -jar payara-micro-4.1.153.jar --rootDir c:\path\`
    `java -jar payara-micro-4.1.153.jar --domainConfig c:\path\domain.xml`
results in the following error:
```
Exception in thread "main" fish.payara.micro.BootstrapException: java.lang.IllegalArgumentException: Illegal character in authority at index 7: file://c:\path\domain.xml
                at fish.payara.micro.PayaraMicro.bootStrap(PayaraMicro.java:712)
                at fish.payara.micro.PayaraMicro.main(PayaraMicro.java:105)
Caused by: org.glassfish.embeddable.GlassFishException: java.lang.IllegalArgumentException: Illegal character in authority at index 7: file://c:\path\domain.xml
                at com.sun.enterprise.glassfish.bootstrap.StaticGlassFishRuntime.newGlassFish(StaticGlassFishRuntime.java:139)
                at fish.payara.micro.PayaraMicro.bootStrap(PayaraMicro.java:692)
                ... 1 more
Caused by: java.lang.IllegalArgumentException: Illegal character in authority at index 7: file://c:\path\domain.xml
                at java.net.URI.create(Unknown Source)
                at org.glassfish.kernel.embedded.EmbeddedDomainXml.getDomainXml(EmbeddedDomainXml.java:73)
                at org.glassfish.kernel.embedded.EmbeddedDomainXml.getDomainXml(EmbeddedDomainXml.java:66)
                at org.glassfish.config.support.DomainXml.run(DomainXml.java:117)
                at org.jvnet.hk2.config.ConfigurationPopulator.populateConfig(ConfigurationPopulator.java:58)
                at org.glassfish.hk2.bootstrap.HK2Populator.populateConfig(HK2Populator.java:106)
                at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.populateConfig(AbstractModulesRegistryImpl.java:211)
                at com.sun.enterprise.module.bootstrap.Main.createServiceLocator(Main.java:273)
                at com.sun.enterprise.glassfish.bootstrap.StaticGlassFishRuntime.newGlassFish(StaticGlassFishRuntime.java:112)
                ... 2 more
Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: file://c:\path\domain.xml
                at java.net.URI$Parser.fail(Unknown Source)
                at java.net.URI$Parser.parseAuthority(Unknown Source)
                at java.net.URI$Parser.parseHierarchical(Unknown Source)
                at java.net.URI$Parser.parse(Unknown Source)
                at java.net.URI.<init>(Unknown Source)
                ... 11 more
```
The reason seems to be, that the results of getAbsolutePath calls in the method PayaraMicro.bootStrap that are passed down as parameter via "setConfigFileURI" are not valid URIs and should be escaped/transformed if OS == Windows.

--- a/nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java
+++ b/nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java
@@ -785,7 +785,7 @@ public class GenericGrizzlyListener implements GrizzlyListener {
                                         final Http2 http2Element,
                                         final FilterChainBuilder builder,
                                         final boolean secure) {
-        if (http2Element == null || http2Element.getEnabled()) {
+        if (http2Element != null && http2Element.getEnabled()) {
 
             // first try to lookup a service appropriate for the mode
             // that has been configured.

--- a/nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/Http2.java
+++ b/nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/Http2.java
@@ -51,7 +51,7 @@ public interface Http2 extends ConfigBeanProxy, PropertyBag {
     final int MAX_CONCURRENT_STREAMS = 100;
     final int INITIAL_WINDOW_SIZE_IN_BYTES = 64 * 1024 - 1;
     int MAX_FRAME_PAYLOAD_SIZE_IN_BYTES = (1 << 24) - 1;
-    final boolean ENABLED = true;
+    final boolean ENABLED = false;
     
     /**
      * Enables HTTP2 support.

--- a/nucleus/grizzly/nucleus-grizzly-all/pom.xml
+++ b/nucleus/grizzly/nucleus-grizzly-all/pom.xml
@@ -137,7 +137,12 @@
         </dependency>
         <dependency>
             <groupId>org.glassfish.grizzly</groupId>
-            <artifactId>grizzly-http-server</artifactId>
+            <artifactId>grizzly-http2</artifactId>
+        </dependency>      
+        <dependency>
+            <groupId>org.glassfish.grizzly</groupId>
+            <artifactId>grizzly-npn-api</artifactId>
+            <version>1.2</version>
         </dependency>
         <dependency>
             <groupId>org.glassfish.main.grizzly</groupId>

--- a/nucleus/payara-modules/payara-micro/src/main/java/fish/payara/micro/PayaraMicro.java
+++ b/nucleus/payara-modules/payara-micro/src/main/java/fish/payara/micro/PayaraMicro.java
@@ -651,7 +651,7 @@ public class PayaraMicro {
 
             if (alternateDomainXML != null) {
                 gfproperties.setConfigFileReadOnly(false);
-                gfproperties.setConfigFileURI("file://" + alternateDomainXML.getAbsolutePath());
+                gfproperties.setConfigFileURI("file:///" + alternateDomainXML.getAbsolutePath().replace('\\', '/'));
             } else {
                 if (noCluster) {
                     gfproperties.setConfigFileURI(Thread.currentThread().getContextClassLoader().getResource("microdomain-nocluster.xml").toExternalForm());
@@ -667,7 +667,9 @@ public class PayaraMicro {
                 if (!configFile.exists()) {
                     installFiles(gfproperties);
                 } else {
-                    gfproperties.setConfigFileURI("file://" + rootDir.getAbsolutePath() + File.separator + "config" + File.separator + "domain.xml");
+                    String absolutePath = rootDir.getAbsolutePath();
+                    absolutePath = absolutePath.replace('\\', '/');
+                    gfproperties.setConfigFileURI("file:///" + absolutePath + "/config/domain.xml");
                     gfproperties.setConfigFileReadOnly(false);
                 }
 
@@ -902,7 +904,7 @@ public class PayaraMicro {
                             + "--domainConfig overrides the complete server configuration with an alternative domain.xml file\n"
                             + "--minHttpThreads the minimum number of threads in the HTTP thread pool\n"
                             + "--maxHttpThreads the maximum number of threads in the HTTP thread pool\n"
-                            + "--hzConfigFile the hazelcast-configuration file to use to override the in-built hazelcast cluster configuration"
+                            + "--hzConfigFile the hazelcast-configuration file to use to override the in-built hazelcast cluster configuration\n"
                             + "--autoBindHttp sets autobinding of the http port to a non-bound port\n"
                             + "--autoBindSsl sets autobinding of the https port to a non-bound port\n"
                             + "--autoBindRange sets the maximum number of ports to look at for port autobinding\n"

